test/python/transpiler/test_1q.py

```diff
@@ -59,8 +59,23 @@ class Test1QWorking(QiskitTestCase):
              level=[0, 1, 2, 3],
              dsc='Transpiling {circuit.__name__} at level {level} should work',
              name='{circuit.__name__}_level{level}_valid')
-    def test(self, circuit, level):
+    def test_device(self, circuit, level):
         """All the levels with all the 1Q backend"""
         result = transpile(circuit(), backend=Fake1Q(), optimization_level=level,
                            seed_transpiler=42)
         self.assertIsInstance(result, QuantumCircuit)
+
+    @combine(circuit=[circuit_3516],
+             level=[0, 1, 2, 3],
+             dsc='Transpiling {circuit.__name__} at level {level} should work for simulator',
+             name='{circuit.__name__}_level{level}_valid')
+    def test_simulator(self, circuit, level):
+        """All the levels with all the 1Q simulator backend"""
+        # Set fake backend config to simulator
+        backend = Fake1Q()
+        config = backend.configuration()
+        config.simulator = True
+        backend._configuration = config
```

Comment:
This would persist in `Fake1Q` for other tests, but Fake1Q needs to be cleaned up post-release anyway.

Reply:
Since its a new object, wont it disappear as soon as it goes out of scope at end of the test?

Reply:
Yup, you're right.